code/renderers/vue3/src/entry-preview-docs.ts

```diff
@@ -1,6 +1,7 @@
-import type { ArgTypesEnhancer, DecoratorFunction } from '@storybook/types';
-import { extractComponentDescription, enhanceArgTypes } from '@storybook/docs-tools';
+import { enhanceArgTypes, extractComponentDescription } from '@storybook/docs-tools';
+import type { ArgTypesEnhancer } from '@storybook/types';
 import { extractArgTypes } from './docs/extractArgTypes';
+import { sourceCodeDecorator } from './docs/source-code-generator';
```

Comment:
I appreciate your approach of preserving the previous code as legacy. However, it seems like your new code could be integrated as an improvement. Could you clarify why it's necessary to retain the legacy code since no major features appear to be overridden?

Reply:
I think my point was about the <script setup> feature. Currently there is an opt-in parameter the user can set to move props to the <script> section.

My goal would be to automatically move complex props (object, arrays and functions) to the <script> section and keep primitive props (string, Boolean, number) inside the <template>.

This is not breaking change on its own but it could break code snippets of projects that use custom source code transformers to e.g. add import statements etc. But if you say that’s not considered a breaking change for Storybook I am totally finde since it’s probably a rare edge case.